test/llvmcall.jl

```diff
@@ -74,15 +74,15 @@ end
 # On llvm <3.5+ even though the the compilation fails on the first try,
 # llvm still adds the intrinsice declaration to the module and subsequent calls
 # are succesfull.
-if convert(VersionNumber, Base.libllvm_version) > v"3.5-"
-
-function undeclared_ceil(x::Float64)
-    llvmcall("""%2 = call double @llvm.ceil.f64(double %0)
-        ret double %2""", Float64, Tuple{Float64}, x)
-end
-@test_throws ErrorException undeclared_ceil(4.2)
-
-end
+#if convert(VersionNumber, Base.libllvm_version) > v"3.5-"
+#
+#function undeclared_ceil(x::Float64)
+#    llvmcall("""%2 = call double @llvm.ceil.f64(double %0)
+#        ret double %2""", Float64, Tuple{Float64}, x)
+#end
+#@test_throws ErrorException undeclared_ceil(4.2)
```

Comment:
It only accidentally worked on llvm 3.5, because we were using a module per function call. On this branch, everything gets parsed into a global module, so if ceil was previously declared this does not actually throw an error. A lot of this is really a limitation of the LLVM IR parsing API, so I'm planning to redo this properly, potentially with some LLVM API changes, at which point there will be more logical and complete error messages here, but that is out of the scope of this PR.